master/internal/trial.go

```diff
@@ -381,6 +381,33 @@ func (t *trial) maybeAllocateTask() error {
 	}
 
 	restoredAllocation, err := t.maybeRestoreAllocation()
+
+	if restoredAllocation == nil {
+		launchWarnings, err := t.rm.ValidateResourcePoolAvailability(
+			&sproto.ValidateResourcePoolAvailabilityRequest{
```

Comment:
can we add another check that we only check the capacity if we have at least one blocked nodes from ```logpattern.GetBlockedNodes(context.TODO(), *msg.TaskID)```

I am somewhat worried that someone might lose a bunch of agents and all their jobs would fail because of this

Reply:
So this only apply to the restarted experiment?

Reply:
I would add the check right before we call ```ValidateResourcePoolAvailability```

Yes so only don't check for the restart case of calling ```ValidateResourcePoolAvailability```.